api/spec/src/entitlements/entitlements.tsp

```diff
@@ -0,0 +1,319 @@
+  ): Entitlement[] | OpenMeter.PaginatedResponse<Entitlement> | OpenMeter.CommonErrors;
+
+  /**
+   * Get entitlement by id.
+   */
+  @get
+  @operationId("getEntitlementById")
+  get(
+    @path
+    entitlementId: ULID,
+  ): Entitlement | OpenMeter.CommonErrors | OpenMeter.NotFoundError;
+}
+
+/**
+ * Type of the entitlement.
+ */
+@friendlyName("EntitlementType")
+enum EntitlementType {
+  metered: "metered",
+  boolean: "boolean",
+  static: "static",
+}
+
+// NOTE: use OmitProperties<T> for templates once entitlement models are defined
+
+/**
+ * Entitlement templates are used to define the entitlements of a plan.
+ * Features are omitted from the entitlement template, as they are defined in the rate card.
+ */
+@discriminator("type")
+@friendlyName("Entitlement")
+union Entitlement {
+  metered: EntitlementMetered,
+  static: EntitlementStatic,
+  boolean: EntitlementBoolean,
+}
+
+/**
+ * Shared fields for entitlement creation
+ */
+@friendlyName("EntitlementCreateSharedFields")
+model EntitlementCreateSharedFields {
+  /**
+   * The feature the subject is entitled to use.
+   * Either featureKey or featureId is required.
+   */
+  @example("example-feature-key")
+  featureKey?: Key;
+
+  /**
+   * The feature the subject is entitled to use.
+   * Either featureKey or featureId is required.
+   */
+  @example("01ARZ3NDEKTSV4RRFFQ69G5FAV")
+  featureId?: ULID;
+
+  /**
+   * Additional metadata for the feature.
+   */
+  metadata?: Metadata;
+
+  usagePeriod?: RecurringPeriod;
+}
+
+/**
+ * Create inpurs for metered entitlement
+ */
+@friendlyName("EntitlementMeteredCreateInputs")
+model EntitlementMeteredCreateInputs {
+  ...OmitProperties<EntitlementCreateSharedFields, "usagePeriod">;
+  type: EntitlementType.metered;
+
+  /**
+   * If softLimit=true the subject can use the feature even if the entitlement is exhausted, hasAccess will always be true.
+   */
+  isSoftLimit?: boolean = false;
+
+  /**
+   * Deprecated, ignored by the backend. Please use isSoftLimit instead; this field will be removed in the future.
+   */
+  @deprecated("Use isSoftLimit instead")
+  isUnlimited?: boolean = false;
+
+  usagePeriod: RecurringPeriod;
+  measureUsageFrom?: MeasureUsageFrom;
+
+  /**
+   * You can grant usage automatically alongside the entitlement, the example scenario would be creating a starting balance.
+   * If an amount is specified here, a grant will be created alongside the entitlement with the specified amount.
+   * That grant will have it's rollover settings configured in a way that after each reset operation, the balance will return the original amount specified here.
+   * Manually creating such a grant would mean having the "amount", "minRolloverAmount", and "maxRolloverAmount" fields all be the same.
+   */
+  @minValue(0)
+  issueAfterReset?: float64;
+
+  /**
+   * Defines the grant priority for the default grant.
+   */
+  @minValue(1)
+  @maxValue(255)
+  issueAfterResetPriority?: uint8 = 1;
```

Comment:
Take a look at how this will be serialized. I always pick `integer` instead of the more fine-grained alternative for this reason.

Reply:
I don't understand your issue. This is the current version:
```yaml
issueAfterResetPriority:
  type: integer
  minimum: 1
  maximum: 255
  default: 1
  example: 1
  description: |
    Defines the grant priority for the default grant. If provided, issueAfterReset must have a value.
```

this is the TypeSpec serialized version:
```yaml
issueAfterResetPriority:
  type: integer
  format: uint8
  minimum: 1
  maximum: 255
  description: Defines the grant priority for the default grant.
  default: 1
```
format is the correct keyword to describe that limitation (we could even do away with max 255 as thats also included in the format)

Reply:
If it works, then ok, it was just a heads-up that sometimes the format isn't correct and a number could be emitted as a string.